ci/test/00_setup_env_native_tsan.sh

```diff
@@ -11,6 +11,11 @@ export CI_IMAGE_NAME_TAG="docker.io/ubuntu:24.04"
 export APT_LLVM_V="19"
 export PACKAGES="clang-${APT_LLVM_V} llvm-${APT_LLVM_V} libclang-rt-${APT_LLVM_V}-dev libc++abi-${APT_LLVM_V}-dev libc++-${APT_LLVM_V}-dev python3-zmq"
 export DEP_OPTS="CC=clang-${APT_LLVM_V} CXX='clang++-${APT_LLVM_V} -stdlib=libc++'"
+export LIBMULTIPROCESS_CMAKE_OPTIONS="-DCMAKE_BUILD_TYPE=Tsan"
```

Comment:
Hmm, I though that only `Debug`, `Release`, `MinSizeRel` and `RelWithDebInfo` are allowed, no? What is this `Tsan` build type doing? `git grep Tsan` in the top directory yields no results.

Reply:
Actually that's the same output as when you don't use `-DCMAKE_BUILD_TYPE=Tsan`.

Reply:
The difference can be seen at the install stage:

```
-- Install configuration: ""
```

vs

```
-- Install configuration: "Tsan"
```

Reply:
https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html

> Typical values include `Debug`, `Release`, `RelWithDebInfo` and `MinSizeRel`, but custom build types can also be defined.

However, I do not think that our build has defined a custom build type `Tsan`. I checked that `-DCMAKE_BUILD_TYPE=Tsan` uses the same compile flags as `-DCMAKE_BUILD_TYPE=FooBarBaz` which are not the same as any of `Debug`, `Release`, `RelWithDebInfo` or `MinSizeRel`.

I think it would be better to drop this and leave it to the default or if needed, explicitly specify any of `Debug`, `Release`, `RelWithDebInfo` or `MinSizeRel`, or if needed actually implement a new build type `Tsan` in `CMakeLists.txt`.

Reply:
This was based on the suggestion by @maflcko:

> You could build libmultiprocess with tsan (edit: in the existing tsan task) to get a better check.

So I guess libmultiprocess needs to explicitly support this?

@ryanofsky thoughts?

Reply:
That should be a matter of compiling libmiltiprocess with `-fsanitize=thread`. Maybe using `cmake -DCMAKE_CXX_FLAGS=-fsanitize=thread` when compiling libmiltiprocess would suffice (untested).

Reply:
Done. Also checked locally (`cmake --build . --verbose`) that this actually uses `-fsanitize=thread`